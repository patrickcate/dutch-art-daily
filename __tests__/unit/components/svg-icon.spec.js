import { shallowMount } from '@vue/test-utils'
import SvgIcon from '@components/svg-icon.vue'

describe('SvgIcon Component', () => {
  it('is a Vue instance', () => {
    const wrapper = shallowMount(SvgIcon, {
      propsData: {
        icon: {
          viewBox: '0 0 10 10',
          url: 'test.svg',
        },
      },
    })
    expect(wrapper.isVueInstance()).toBe(true)
  })

  it('renders correctly', () => {
    const wrapper = shallowMount(SvgIcon, {
      propsData: {
        icon: {
          viewBox: '0 0 10 10',
          url: 'test.svg',
        },
      },
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('renders with custom width and height', () => {
    const wrapper = shallowMount(SvgIcon, {
      propsData: {
        icon: {
          viewBox: '0 0 10 10',
          url: 'test.svg',
        },
        width: '20px',
        height: '20px',
      },
    })

    expect(wrapper).toMatchSnapshot()
  })
})
