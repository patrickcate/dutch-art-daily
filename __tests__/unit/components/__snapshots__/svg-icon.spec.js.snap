// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SvgIcon Component renders correctly 1`] = `
<svg viewBox="0 0 10 10" width="1em" height="1em" class="svg-icon">
  <use xlink:href="test.svg"></use>
</svg>
`;

exports[`SvgIcon Component renders with custom width and height 1`] = `
<svg viewBox="0 0 10 10" width="20px" height="20px" class="svg-icon">
  <use xlink:href="test.svg"></use>
</svg>
`;
